config/environments/production.rb

```diff
@@ -91,6 +92,25 @@
       password: ENV['SMTP_PASSWORD'],
       authentication: ENV['SMTP_AUTH'],
       enable_starttls_auto: ENV['SMTP_STARTTLS_AUTO'],
+      openssl_verify_mode: ENV['SMTP_OPENSSL_VERIFY_MODE'],
+    }
+  elsif ENV['SMTP_AUTH'].present? && ENV['SMTP_AUTH'] != "none"
```

Comment:
Is there a need for all of these if statements? Based on this (https://guides.rubyonrails.org/action_mailer_basics.html), there is only 2 options `none` or `peer`. You can just set `openssl_verify_mode: ENV['SMTP_OPENSSL_VERIFY_MODE'].presence || "peer"` and remove all the elsif statements

Reply:
or you could just add it as the last line with `openssl_verify_mode: ENV['SMTP_OPENSSL_VERIFY_MODE'] if ENV['SMTP_OPENSSL_VERIFY_MODE'].present?`

Reply:
With peer it expects a locally installed certificate. The 'default' options seems to be that this constant is not set in any way.
I tried several ways of using an if statement in this, but they all failed for me. Including your second suggestion, sadly
Also it can't be an empty string as it then complains about the uninitialized constant.

But please don't misunderstand me, the current solution does not make me very happy either and i'm very happy about any input

Reply:
Oh I see. In that case, you can just add this line at the very end `ActionMailer::Base.smtp_settings[:openssl_verify_mode] = ENV['SMTP_OPENSSL_VERIFY_MODE'] if ENV['SMTP_OPENSSL_VERIFY_MODE'].present?` since the smtp_settings is just a Rails hash

Reply:
This one does work, very nice. Thanks